### PR TITLE
Support h5pyd

### PR DESCRIPTION
**Pull request by vallsv**
Support `h5pyd` only on the notebook part.

- Create abstraction to check object types
- Use `__module__` from provided object to check if the library is known. If the library is `h5pyd`, register it's class objects.

I use `h5serv` with it's default configuration and file
```
git clone https://github.com/HDFGroup/h5serv.git
cd h5serv
python h5serv
```

Install `h5pyd`
```
pip install h5pyd
```

I checked using a notebook

Screenshot from 2019-04-16 15-09-38

### CODE REVIEW

**Comment by vallsv** [comment]
Oh, BTW, the file i tested contain an external dataset (`extlink`) which is returned by `h5pyd` as a None dataset (broken link maybe). Then it is not possible to reach the full path from the dataset.

Then i hide the link to copy the path.

It could be fixed by providing the full path to `item_for_dataset` (computing it using each consecutive keys). I think it is also more accurate cause it is the exact path you use to select the dataset, despite soft links. If you like i can update it in this way.

File: h5glance/utils.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    lib = obj_class.__module__.split(".")[0]
+    if lib == "h5pyd":
+        register_h5pyd()
+    else:
+        _mapping[obj_class] = None
+
+
+def is_file(obj):
+    """Returns true if the object is a h5py-like file."""
+    obj_class = obj.__class__
```
Comment (takluyver) [inline h5glance/utils.py]: I think the idiomatic way to get the class of an object is now `type(obj)`.

Comment (vallsv) [reply h5glance/utils.py]: It's up to you.

I update the code and create a function `get_h5py_kind` to make it simple.

**Comment by vallsv** [comment]
> I have some concern that looking up the class in a dict rather than using `isinstance()`

Yes you are right. I can fix that into another proposal if you like.

File: h5glance/utils.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"""Helper functions
+"""
+import h5py
+
+_mapping = {}
+_mapping[h5py.File] = "file"
+_mapping[h5py.Group] = "group"
+_mapping[h5py.Dataset] = "dataset"
```
Comment (takluyver) [inline h5glance/utils.py]: Why not write the initial mapping with these in?

```python
_mapping = {
    h5py.File: 'file',
...
```

Comment (vallsv) [reply h5glance/utils.py]: It's up to you. I updated as you requested.

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**